### PR TITLE
Implement pseudo-grid movement

### LINKED ISSUE #12: Implement grid-based movement

### LINKED ISSUE #26: Collision detection
Fix the bounding box of the character's sprite

### PR DESCRIPTION
**Pull request by pparuzel**
Resolves #12 
Resolves #26 
Pixel perfect grid movement is impossible using this version of Phaser unfortunatelly.

### CODE REVIEW

File: src/player.js [hunk 1]
```diff
@@ -3,38 +3,88 @@ class Character {
...
+        this._blocked = null;  // initialized with Sprite
     }
 
     attachSprite(sprite) {
         this.sprite = sprite;
         // Normalize and scale the velocity so that player can't move faster along a diagonal
         this.sprite.body.velocity.normalize().scale(this._speed);
         this.sprite.setCollideWorldBounds(true);
+        this.sprite.body.setSize(16, 8);
+        this.sprite.body.setOffset(8, 24);
```
Comment (pparuzel) [inline src/player.js:21]: Lines 20 and 21 resolve collision problems

**Review by cinekqw1 — APPROVED** [review]
ok

File: src/player.js [hunk 2]
```diff
@@ -3,38 +3,88 @@ class Character {
...
+            } else {
+                this.sprite.anims.stopOnRepeat();
+            }
         }
+        this._prevPosition.x = this.sprite.x;
+        this._prevPosition.y = this.sprite.y;
+    }
+}
+
+function is_between(expected, xval, yval, error=0) {
```
Comment (fturek) [inline src/player.js]: I believe that we should use cammelCase here. Also for the parameters.

File: src/player.js [hunk 3]
```diff
@@ -3,38 +3,88 @@ class Character {
...
-        } else {
-            this.sprite.anims.stopOnRepeat();
+        this._handleMovement();
+    }
+
+    isBlocked() {
+        return this._blocked.down || this._blocked.up || this._blocked.left || this._blocked.right;
+    }
+
+    _moveWithStep(key, stepSizeX, stepSizeY) {
```
Comment (fturek) [inline src/player.js]: What means underscore sign before methods names?? Do we need them?

File: src/player.js [hunk 4]
```diff
@@ -3,38 +3,88 @@ class Character {
         this.name = name;
         this.hp = 100;
         this._speed = 100;
+        this._isMoving = false;
```
Comment (fturek) [inline src/player.js]: Do we need underscore sign between some variables names? There is no consistency now.

**Review by fturek — COMMENTED** [review]
I have left some comments here.

File: src/player.js (hunk 2 above)
Comment (pparuzel) [reply src/player.js]: sure

File: src/player.js (hunk 3 above)
Comment (pparuzel) [reply src/player.js]: _ before a function means it's "private" or in other words: not public interface or in other words:
avoid changing this functions code :)

Comment (fturek) [reply src/player.js]: ok, I see. We can follow this convection.

**Comment by pparuzel** [comment]
I removed the "underscores convention" to improve understanding of the code for us all.
Also corrected constants which were using it as well.

**Pull request merged**